src/ts/fileDescriptorTSServices.ts

```diff
@@ -72,7 +77,151 @@ export function printFileDescriptorTSServices(fileDescriptor: FileDescriptorProt
+        service,
+        method,
+        camelCaseMethodName,
+        requestMessageTypeName,
+        responseMessageTypeName
+      );
+    }
+  });
+  printer.dedent().printLn("}");
+}
+
+function printUnaryStubMethod(
+  printer: CodePrinter,
+  service: ServiceDescriptorProto,
+  method: MethodDescriptorProto,
+  camelCaseMethodName: string,
+  requestMessageTypeName: string,
+  responseMessageTypeName: string
+) {
+  printer
+             .printLn(`${camelCaseMethodName}(`)
+      .indent().printLn(`requestMessage: ${requestMessageTypeName},`)
+               .printLn(`metadata?: grpc.Metadata,`)
+               .printLn(`callback?: (error: any, responseMessage: ${responseMessageTypeName}|null) => void`)
+    .dedent().printLn(`): void {`)
+      .indent().printLn(`grpc.unary(${service.getName()}.${method.getName()}, {`)
+        .indent().printLn(`request: requestMessage,`)
+                 .printLn(`host: this.serviceHost,`)
+                 .printLn(`metadata,`)
+                 .printLn(`onEnd: (response: grpc.UnaryOutput<${responseMessageTypeName}>): void => {`)
+          .indent().printLn(`if (callback) {`)
+            .indent().printLn(`const responseMessage = response.message;`)
+                     .printLn(`if (response.status !== grpc.Code.OK) {`)
+              .indent().printLn(`return callback(response, null);`)
+            .dedent().printLn(`} else {`)
+              .indent().printLn(`callback(null, responseMessage);`)
+            .dedent().printLn(`}`)
+          .dedent().printLn(`}`)
+        .dedent().printLn(`}`)
+      .dedent().printLn(`});`)
+    .dedent().printLn(`}`);
+}
+
+function printServerStreamStubMethod(
+  printer: CodePrinter,
+  service: ServiceDescriptorProto,
+  method: MethodDescriptorProto,
+  camelCaseMethodName: string,
+  requestMessageTypeName: string,
+  responseMessageTypeName: string
+) {
+  printer
+           .printLn(`${camelCaseMethodName}(requestMessage: ${requestMessageTypeName}, metadata?: grpc.Metadata): any {`)
+    .indent().printLn(`const listeners: {`)
+      .indent().printLn(`data: Array<(response: ${responseMessageTypeName}) => void>,`)
+               .printLn(`end: Array<() => void>`)
+    .dedent().printLn(`} = {`)
+      .indent().printLn(`data: [],`)
+               .printLn(`end: []`)
+    .dedent().printLn(`};`)
+             .printLn(`grpc.invoke(${service.getName()}.${method.getName()}, {`)
+      .indent().printLn(`request: requestMessage,`)
+               .printLn(`host: this.serviceHost,`)
+               .printLn(`metadata,`)
+               .printLn(`onMessage: (responseMessage: ${responseMessageTypeName}) => {`)
+        .indent().printLn(`listeners.data.forEach(callback => {`)
+          .indent().printLn(`callback(responseMessage);`)
+        .dedent().printLn(`});`)
+               .printLn(`},`)
+               .printLn(`onEnd: () => {`)
+          .indent().printLn(`listeners.end.forEach(callback => {`)
+                   .printLn(`callback();`)
```

Comment:
Should we clear the listeners at this point to avoid garbage issues with referenced callbacks?